client.go

```diff
@@ -58,6 +89,17 @@ func NewClient(apiURL string, accessToken string) *Client {
 	return &client
 }
 
+// Utilizing ClientAPI Interface
+func NewClientAPI(apiURL string, accessToken string) ClientAPI {
+	client := Client{
```

Comment:
We can remove this client variable and directly return like this 
return  &Client{
          APIURL:  apiURL,
        AccessToken: accessToken,
       HTTP:  cleanhttp.DefaultClient(),
}